source/components/composing-components.md

```diff
@@ -0,0 +1,260 @@
+We can consume it in the following way:
+
+```app/users/template.hbs
+{{#user-list users=users as |user|}}
+  {{user-card user=user}}
+{{/user-list}}
+```
+
+Now `{{user-card}}` has access to the current user, which changes as the `{{user-list}}` iterates over each user
+since the `{{yield user}}` is inside the `{{each}}` block. Although this is nice, what if we used our component
+without a block, i.e. `{{user-list users=users}}`? This would make the component pretty useless since nothing is yielding,
+but the users are still being iterated. Let's mix in the `hasBlock` attribute and see if we can make it more useful.
+
+```app/components/user-list/template.hbs
+{{#if hasBlock}}
+  {{#each users as |user|}}
+    {{yield user}}
+  {{/each}}
+{{else}}
+  {{#each users as |user|}}
+    <section class="user-info">
+      <h3>{{user.name}}</h3>
+      <p>{{user.bio}}</p>
+    </section>
+  {{/each}}
+{{/if}}
+```
+
+Now we our component is a little more useful, because it has sane defaults, but it also allows us to override
+those defaults. I'm sure there is more that we can do to our `{{user-list}}` component to make it more useful,
+like handling different types of users.
+
+#### `{{component}}` Helper
+
+Now we need to add a new type of user, a superuser, which will have different behavior in our
+application. How would the consumer use this new data to show a different UI for each type of user?
+This is where the `{{component}}` helper comes into play. This helper allows us to render a component
+based on an arbitrary value. Along with this new feature request, we were also asked to add a toggle
+between a basic and a detailed view, which we need to expose in our block.
+
+Since components require a dash in the component name, e.g. `basic-card-superuser`,
+the simplest way would be to use a helper to concatenate 'basic-card-' and `user.type`.
+
+```app/users/template.hbs
+{{#user-list users=users as |user basicMode|}}
+  {{#if basicMode}}
+    {{component (concat 'basic-card-' user.type) user=user}}
+  {{else}}
+    {{component (concat 'detailed-card-' user.type) user=user}}
+  {{/if}}
+{{/user-list}}
+```
+
+Now we have `{{basic-card-superuser}}` and `{{detailed-card-public}}` rendering in their respective places
+based on the `user.type` equaling 'superuser' and 'public' respectively. Here we used the 'concat' helper
+as a subexpression inside of our `{{component}}` helper. Subexpressions are evaluated before the helper
+and then the helper can use the value, in this case it's just a concatenation of two values. You can
+also experiment with other helpers as subexpressions, like the `if` helper.
+
+Apart from the use of `{{component}}`, we also have the second yielded value for our basic/detailed mode feature,
+which we've added to our yield, i.e. `{{yield user basic}}`.
+
+Something to remember is that the yielded values are ordered sequentially by how they are defined, so you
```

Comment:
I think you mention that they are positional parameters above when you introduce valid usages of `{{yield}}` maybe that introduction is a better place to make clear you can pass any number of params.